tests/test_stl.py

```diff
@@ -46,6 +46,13 @@ def test_array(doc):
     )
 
 
+def test_array_no_default_ctor():
+    lst = m.NoDefaultCtorArray(3)
+    assert [e.val for e in lst.arr] == [13, 23]
```

Comment:
Do you allow mutation of members of the array? if so might want to add a test about whether mutating lst.arr affects m.NoDefaultCtorArray(4).arr.

Reply:
The `array_caster` copies (or moves) elements to/from a native Python `list`. The code below demonstrates the behavior. I verified that it works, then undid the change again, because that aspect of the `array_caster` behavior is tangential to adding the default ctor support.

```python
def test_array_no_default_ctor():
    lst = m.NoDefaultCtorArray(3)
    assert [e.val for e in lst.arr] == [13, 23]
    lst4 = m.NoDefaultCtorArray(4)
    lst.arr = lst4.arr
    assert [e.val for e in lst.arr] == [14, 24]
    # Tangential to default ctor behavior:
    lst.arr = m.NoDefaultCtorArray(5).arr
    assert [e.val for e in lst.arr] == [15, 25]
    assert [e.val for e in lst4.arr] == [14, 24]
```